src/client.rs

```diff
@@ -161,12 +162,11 @@ pub fn run(args: &::clap::ArgMatches, address: &str) {
             } else {
                 prompt = "sonnerie> "
             };
-            let r = rl.readline(prompt);
-            let line;
-            match r {
+
+            let line = match rl.readline(prompt) {
```

Comment:
`match` and `if` are expressions, so we can assign the return value directly from the result of the expression